src/entities/ensRegistration.ts

```diff
@@ -0,0 +1,23 @@
+import { EthereumAddress } from '.';
+import { ENS_RECORDS } from '@rainbow-me/helpers/ens';
+
+export type Records = { [key in keyof typeof ENS_RECORDS]: string };
+
+export interface RegistrationParameters {
+  name: string;
+  duration: number;
+  records: Records;
+  ownerAddress: EthereumAddress;
+  rentPrice: string;
+  salt?: string;
+  setReverseRecord?: boolean;
+}
+
+export interface ENSRegistrationState {
+  currentRegistrationName: string;
+  registrations: {
+    [key: EthereumAddress]: {
```

Comment:
I feel like we don't need this? Shouldn't the ENS name as the key itself be sufficient enough? Since we aren't scoping `currentRegistrationName` under an account address, feels like it's not dependent on it. It would reduce a lot of the complexity in the redux reducer as well